package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -146,13 +150,19 @@ def _prepare(self):
         group = getattr(self._ags[0], self.grouping)
 
         # Get masses and charges for the selection
-        try:  # in case it's not an atom
-            self.masses = np.array([elem.total_mass() for elem in group])
-            self.charges = np.array([elem.total_charge() for elem in group])
-        except AttributeError:  # much much faster for atoms
+        if self.grouping == "atoms":
             self.masses = self._ags[0].masses
             self.charges = self._ags[0].charges
 
+        elif self.grouping in ["residues", "segments", "fragments"]:
+            self.masses = np.array([elem.atoms.total_mass() for elem in group])
```

Comment:
The method `group.total_mass` and `total_charge` have a parameter for `compound` to provide grouping which is faster than list compregensions. Is there a reason not to use it here?